independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/ListExtensionsCommandHandler.java

```diff
@@ -86,28 +98,69 @@ public QuarkusCommandOutcome execute(QuarkusCommandInvocation invocation) throws
         } catch (IOException e) {
             throw new QuarkusCommandException("Failed to determine the list of installed extensions", e);
         }
+
+        Predicate<Extension> categoryFilter;
+        if (StringUtils.isNotBlank(category)) {
+            categoryFilter = e -> ExtensionProcessor.of(e).getCategories().contains(category);
+        } else {
+            categoryFilter = e -> true;
+        }
+
         extensions.stream()
                 .filter(e -> !ExtensionProcessor.of(e).isUnlisted())
+                .filter(categoryFilter)
+                .sorted(Comparator.comparing(e -> e.getArtifact().getArtifactId()))
                 .forEach(e -> display(log, e, installedByKey.get(toKey(e)), all, installedOnly, currentFormatter));
         final BuildTool buildTool = invocation.getQuarkusProject().getBuildTool();
         boolean isGradle = BuildTool.GRADLE.equals(buildTool) || BuildTool.GRADLE_KOTLIN_DSL.equals(buildTool);
 
         if (!cli) {
-            if ("concise".equalsIgnoreCase(format)) {
+            if ("concise".equalsIgnoreCase(format) || StringUtils.isBlank(format)) {
```

Comment:
I would rethink the cli/batchMode sections here (relating to the other comment, too), and push them back into the Gradle/Maven plugin to emit help appropriately (when not coming via the cli).